Remove more Notification variants

djc opened the pull request:
Continuing the work started in

- #4499

djc committed:
notifications: log directly when using download backends

djc committed:
notifications: log directly for valid checksums

djc committed:
notifications: log directly when component is already installed

djc committed:
notifications: log directly when failing to update hash file

djc committed:
notifications: log directly when hash file not found

djc committed:
notifications: log directly after failing to determine memory limit

djc committed:
notifications: log directly on missing components

djc committed:
notifications: log directly when installing components

djc committed:
notifications: log directly when removing components

djc committed:
notifications: log directly for manifest downloads

djc committed:
notifications: log directly for downloaded manifests

djc committed:
notifications: log directly when downloading legacy manifests

djc committed:
notifications: log directly on missing components

djc committed:
notifications: log directly when skipping components

djc committed:
notifications: log directly when removing stray hash files

djc committed:
notifications: log directly when downloading files

djc committed:
notifications: log directly when resuming partial downloads

djc committed:
notifications: log directly when setting auto install mode

djc committed:
notifications: log directly when setting the default toolchain

rami3l commented on tests/suite/cli_v2.rs:
```diff
@@ -1630,7 +1630,7 @@ async fn add_target_again() {
         .await
         .with_stderr(snapbox::str![[r#"
-info: component 'rust-std' for target '[CROSS_ARCH_I]' is up to date
+info: component is up to date component=rust-std target=[CROSS_ARCH_I]
```
Are you sure about this? Message changes like this are making the readability worse, thus this might be a UX regression when it comes to `info!()`, `warn!()` and `error!()`...

rami3l replied:
Or, are you moving towards the machine-readable output this way? If that's the case I'm afraid we need to update our user-facing message format:

https://github.com/rust-lang/rustup/blob/7365ab767a586c13f4c48e89ef18664c0705b37b/src/cli/log.rs#L112

... to make it clearer, say "info: component is up to date **for** component=rust-std target=[CROSS_ARCH_I]" to stay closer to the original vibe.

djc replied:
Yeah, I was hoping to make this more like structured logging. However, I agree that this is a bit of a UX regression, so I'll revert some of the message changes to be more human-friendly.

rami3l replied:
@djc Now that you've brought this up, I think there are actually two ways of doing https://github.com/rust-lang/rustup/issues/450. The first is giving the notification system an alternative output format (corrosion maintainer specifically asked for JSON for better CMake compatibility, so the resulting output would be JSONL), the second would be returning a specifically-designed single JSON value.

If we are moving on with the first option then your work of extracting field names is actually valuable (as I can then further abuse `tracing_subscriber` to output JSON directly), so that is why I proposed the compromise above :)

What do you think about this?

cc @ChrisDenton

ChrisDenton replied:
I find the first option appealing as it seems much more generally useful and I think it's more in line with what tools such as cargo provide. I'm less sure about hacking around `tracing_subscriber` to provide it but I'd want a better idea of what that would look like. Maybe it's ok.

rami3l replied:
@chrisdenton Back in the v1.28 release cycle I've reproduced the regular rustup log messages with `tracing_subscriber` while retaining the previous format, and almost everyone should be using it now without major issues (some aren't even aware of the fact?). 

My option 1 is just a natural extension of that idea.

Of course, special care must be taken to handle the direct prints in say `rustup show`, but I consider that to be a minority.

djc replied:
I don't really want to increase the scope of this PR beyond what it already does, so I think I'll just revert to more human-readable messages. IMO the UX is definitely more important than a currently unimplemented potential future machine-readable mechanism.

djc committed:
chore: avoid trailing whitespace in error message

This messed with the tests, where my editor was chopping of trailing
whitespace. While I'm in here, don't capitalize and parenthesize to
make this line fit in better with the rest of our logging/errors.

rami3l reviewed (approved):
LGTM! I'm still not sure if anyone is relying on parsing our output though since the format has changed, maybe we have to go through a full release to know that.

rami3l commented on src/dist/manifestation.rs:
```diff
@@ -463,12 +479,7 @@ impl Manifestation {
-            &self.target_triple,
-            Some(&self.target_triple),
-        ));
+        info!(component = "rust", "installing component");
```
This will be covered by #4471, not sure if it should be added back in that PR during rebase.

cc @FranciscoTGouveia
